src/main/java/com/scriptbasic/executors/commands/CommandCase.java

```diff
@@ -0,0 +1,137 @@
+import com.scriptbasic.interfaces.BasicRuntimeException;
+import com.scriptbasic.interfaces.Expression;
+import com.scriptbasic.spi.Interpreter;
+import com.scriptbasic.spi.RightValue;
+
+public class CommandCase
+	extends AbstractCommandSelectPart
+{
+	public static interface CaseCondition {
+
+		boolean matchCase(Interpreter interpreter, RightValue selectExpressionValue) throws ScriptBasicException;
+		
+	}
+	
+	public static class EqualCaseCondition implements CaseCondition {
+		final Expression expression;
+		
+		public EqualCaseCondition(final Expression expression) {
+			this.expression = expression;
+		}
+
+		@Override
+		public boolean matchCase(Interpreter interpreter, RightValue selectExpressionValue) throws ScriptBasicException {
+			RightSideEqualsOperator operator = new RightSideEqualsOperator(selectExpressionValue);			
+			operator.setRightOperand(expression);
+			RightValue conditionValue = operator.evaluate(interpreter);
+	        if (conditionValue instanceof AbstractPrimitiveRightValue<?>) {
+	            return BasicBooleanValue.asBoolean(conditionValue);
+	        } else {
+	            throw new BasicRuntimeException(
+	                    "Case condition can not be evaluated to boolean");
+	        }
+		}
+	}
+	
+	public static class FromToCaseCondition implements CaseCondition {
+		final private Expression fromExpression;
+		final private Expression toExpression;
+
+		public FromToCaseCondition(final Expression fromExpression, final Expression toExpression)
+		{
+			this.fromExpression = fromExpression;
+			this.toExpression = toExpression;
+		}
+
+		@Override
+		public boolean matchCase(Interpreter interpreter, RightValue selectExpressionValue) 
+				throws ScriptBasicException {
+			RightSideGreaterOrEqualOperator leftOperator = new RightSideGreaterOrEqualOperator(selectExpressionValue);
+			leftOperator.setRightOperand(fromExpression);
+			RightValue conditionValue = leftOperator.evaluate(interpreter);
+	        if (conditionValue instanceof AbstractPrimitiveRightValue<?>) {
+	            if(!BasicBooleanValue.asBoolean(conditionValue)) {
+	            	return false;
+	            }
+	            // evaluate to condition
+	            RightSideLessOrEqualOperator rightOperator = new RightSideLessOrEqualOperator(selectExpressionValue);
+	            rightOperator.setRightOperand(toExpression);
+	            conditionValue = rightOperator.evaluate(interpreter);
+		        if (conditionValue instanceof AbstractPrimitiveRightValue<?>) {
+		        	return BasicBooleanValue.asBoolean(conditionValue);
+		        }
+	        }
+
+	        throw new BasicRuntimeException("Case condition can not be evaluated to boolean");
+		}
+		
+	}
+	
+	/**
+	 * List of case conditions to be evaluated
+	 */
+	private List<CaseCondition> expressionList = new ArrayList<>();
+	private CommandEndSelect commandEndSelect;
+
+	@Override
+	public void execute(Interpreter interpreter) throws ScriptBasicException {
+		// run this command

```

Comment:
If you feel that this is not documented properly (probably you are right) then extend the JavaDoc on the class `com.scriptbasic.interfaces.Executor` and try to avoid in-line comments.

When adding or extending JavaDoc comments, do in a separate pull request. They will be accepted much faster and easier.